### ISSUE TITLE
hi can you help We bought a plugin for Statamic from you, installed it, and we get a 400 error. The deelp key was bought and installed. What could be the problem?

### ISSUE THREAD

**Issue by ffrostqs** [appswithlove/statamic-one-click-content-translation#10]
hi can you help We bought a plugin for Statamic from you, installed it, and we get a 400 error. The deelp key was bought and installed. What could be the problem? Statamic 4.18.0

![Screenshot 2023-08-21 at 15 31 30](https://github.com/appswithlove/statamic-one-click-content-translation/assets/56931814/47a92cca-c337-4364-8126-5d13f49bdfdd)

**Comment by OleksiiBrylin** [comment]
Do you get the error when you click a translate button or save entry button?

**Comment by ffrostqs** [comment]
when click a translate button

**Comment by OleksiiBrylin** [comment]
@ffrostqs Do you use proper language key for Portugal? 
Could you send a config/statamic/sites.php file content?

**Comment by ffrostqs** [comment]
```
    'sites' => [

        'default' => [
            'name' => 'Kazakhstan',
            'locale' => 'kz_KZ',
            'url' => env('APP_URL').'kz/',
        ],

        'en' => [
            'name' => 'English',
            'locale' => 'en_US',
            'url' => env('APP_URL'),
        ],

        'ua' => [
            'name' => 'Українська',
            'locale' => 'uk_UA',
            'url' => env('APP_URL').'ua/',
        ],

        'es' => [
            'name' => 'Spanish',
            'locale' => 'es_ES',
            'url' => env('APP_URL').'es/',
        ],
        'pt' => [
            'name' => 'Portugal',
            'locale' => 'pt_PT',
            'url' => env('APP_URL').'pt/',
        ],
        
    ],
```

**Comment by ffrostqs** [comment]
the first language is russian, but with a kz tag

**Comment by ffrostqs** [comment]
![image](https://github.com/appswithlove/statamic-one-click-content-translation/assets/56931814/fdcc9dbd-b824-4956-8af7-0219cfa4205e)

**Comment by ffrostqs** [comment]
After update

**Comment by OleksiiBrylin** [comment]
It can be a problem that you have an unavailable source language (locale: kz). 
Please check Deepl documentation: https://www.deepl.com/ru/docs-api/translate-text -> Request Parameters -> source_lang

Also it is not available option to translate on kz.

It means that your default language can't be Kazakhstan

**Comment by OleksiiBrylin** [comment]
@ffrostqs in the last 4.18.1 version I've added an additional config `'ignore_source_lang' => false,`
if you switch it to **true**, the source lang will be used in auto-detection mode

To publish config file
`php artisan vendor:publish --tag="statamic-one-click-content-translation-config"`

**Comment by ffrostqs** [comment]
<img width="474" alt="Screenshot 2023-08-22 at 14 10 44" src="https://github.com/appswithlove/statamic-one-click-content-translation/assets/56931814/3438ad4f-ed4b-46fc-810c-d3ccb2909cad">
but the error remained

**Comment by OleksiiBrylin** [comment]
Have you cleared cache of your browser and statamic?

**Comment by ffrostqs** [comment]
Of course

**Comment by OleksiiBrylin** [comment]
Before the last fix v4.18.1 I saw the issue. 
But with the last fix I can't reproduce the issue.

Statamic v4.18.0
Addon 4.18.**1**
config/statamic-one-click-content-translation.php -> **ignore_source_lang** = **true**
config/statamic/sites.php -> sites.default.locale = kz_KZ;  sites.pt.locale = pt_PT

Remove and install the addon should help in this case

**Comment by ffrostqs** [comment]
restarted composer and reset the cache again and it worked, thanks
